Feature/164 stepper component

qquentinv opened the pull request.

Commit:
Feature 164 : stepper component
- add stepper page
- add stepper in form in nav bar
- add stepper composant not functionnal

Commit:
Feature 164 : Stepper component
- add theme on component

Commit:
Merge branch 'develop' of https://github.com/attineos/atti-components into feature/164_stepper_component

Commit:
Feature 164 : stepper component
- add the necessary element of style

Commit:
Feature 164 : stepper component

Commit:
Feature 164 : stepper component

Commit:
Merge branch 'develop' of https://github.com/attineos/atti-components into feature/164_stepper_component

Commit:
feature 164 : stepper component

Commit:
Feature 164 : stepper component
- add automatic bar addition

Commit:
Feature 164 : stepper component
- add hooks

Commit:
feature 164 : stepper component
- add functional list

Commit:
feature 164 : stepper component
- add functional stepper

Commit:
feature 164 : stepper component
- change bar

qquentinv committed:
Delete StyledEllipse.js

qquentinv committed:
Update index.js

VictorGosse commented on packages/atti-components/src/components/Stepper/hooks/index.js:
```diff
@@ -0,0 +1,53 @@
+}
+
+const StepperProvider = ({ children }) => {
+  const [list, dispatch] = useReducer(stepReducer, {}) // ici pour obtenir un state
```
Comment in English please :)

VictorGosse commented on packages/atti-components/src/components/Stepper/Readme.mdx:
````diff
@@ -0,0 +1,32 @@
+## Examples
+
+```js
+<div>
````
I'm not sure you need this `<div>` in your example.

VictorGosse commented on packages/atti-components/src/components/Stepper/styles/StyledStep.js:
```diff
@@ -0,0 +1,78 @@
+}
+
+Step.propTypes = {
+  id: PropTypes.string.isRequired,
```
Please add some comment to know why we need it.

VictorGosse commented on packages/atti-components/src/components/Stepper/styles/StyledStep.js:
```diff
@@ -0,0 +1,78 @@
+  children: PropTypes.any,
+}
+
+export default Step
```
As the step is clickable, we could imagine that the users will bind some event on it. For example, a click on a step will hide / display a new content on the page.
So you have to add a way for users to add an event on each Step.

VictorGosse commented on packages/atti-components/src/components/Stepper/index.js:
```diff
@@ -0,0 +1,33 @@
+Stepper.Label = StyledLabel
+
+/** @component */
+export default styled(Stepper)``
```
The DOM / accessibility of your component can be improved. You can find some tips here  https://stackoverflow.com/a/52935539
If you need help, don't hesitate to come to me.

VictorGosse commented on packages/atti-components/src/components/Stepper/index.js:
```diff
@@ -0,0 +1,33 @@
+Stepper.Label = StyledLabel
+
+/** @component */
+export default styled(Stepper)``
```
Responsive can be improved too. There are multiple possibilities. Here are some examples:
https://codepen.io/5t3ph/pen/wvwbNqx
https://material.angular.io/components/stepper/overview
On the last one, the stepper becomes vertical. IMO it's the easiest / best solution even if we have a huge number of items.

qquentinv replied:
The stepper is now responsive, thanks for your help :)

VictorGosse commented on packages/atti-components/src/components/Stepper/index.js:
```diff
@@ -0,0 +1,33 @@
+}
+
+Stepper.defaultProps = {
+  variant: 'number',
```
Do you really need the `variant` ?

qquentinv replied:
No, we don't need it :)

VictorGosse commented on packages/atti-components/src/components/Stepper/styles/StyledStep.js:
```diff
@@ -0,0 +1,78 @@
+    top: 0%;
+    width: 200%;
+    margin-top: 50%;
+    border-bottom: 3px solid grey;
```
Should be customizable through the theme.

VictorGosse commented on packages/atti-components/src/components/Stepper/styles/StyledStep.js:
```diff
@@ -0,0 +1,78 @@
+
+  border-radius: ${({ theme }) => theme.components.stepper.sizes.borderRadius};
+  cursor: pointer;
+  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.2);
```
We have a box-shadow in the theme. Maybe you can use it ?

qquentinv replied:
Yeah, I'm gonna use it.

VictorGosse commented on packages/atti-components/src/components/Stepper/styles/StyledLabel.js:
```diff
@@ -0,0 +1,13 @@
+import styled from 'styled-components'
+
+const StyledLabel = styled.p`
```
I don't really like the API of the component. For example this Label that we have to set 2 times and depending on the `middle` prop, it'll be only white.
Moreover the design works right now because you're "lucky", the margin top + font-size of your Label makes it center but if we change the size of the Step (width / height), it won't be centered anymore. You'd better center it on purpose, for example with [flexbox](https://www.w3schools.com/css/css3_flexbox.asp).

For the API, I think that the Label should be only the text that is under the Step. The content of the Step (here a number) can easily be a props (String or Node ?). If you have other ideas, I'm totally opened ;)

qquentinv replied:
I have taken your advice into account and now the internal label is defined by the props value and the external label by a usable tag.
Is that okay?

VictorGosse replied:
👌 It's way better ! Good job !

Commit:
Feature 164 : stepper component

Commit:
Merge branch 'feature/164_stepper_component' of https://github.com/attineos/atti-components into feature/164_stepper_component

Commit:
Feature 164 : Stepper component
- adjustment following the comment

Commit:
Merge branch 'develop' of https://github.com/attineos/atti-components into feature/164_stepper_component

Commit:
feature 164 : stepper component

Commit:
feature 164 : stepper component
- added the possibility for the user to choose the starting step
- modifying the example

Commit:
feature 164 : stepper component
- style rework

Commit:
feature 164 : stepper component
- add accessibility
- add responsive
-  moving the stepper in the navigation menu
- Adding automatic hr adaptation (in progress)

Commit:
feature 164 : stepper component

VictorGosse reviewed (approved):
Well done !
